internal/worker/sshserver/k8s/session.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package k8s
+
+import (
+	"context"
+
+	"github.com/gliderlabs/ssh"
+	"github.com/juju/errors"
+	k8sexec "github.com/juju/juju/caas/kubernetes/provider/exec"
+	"github.com/juju/juju/rpc/params"
+)
+
+// SessionHandler proxies a user's SSH session to a K8s container.
+func (h *Handlers) SessionHandler(session ssh.Session) {
+	handleError := func(err error) {
+		h.logger.Errorf("proxy failure: %v", err)
+		_, _ = session.Stderr().Write([]byte(err.Error() + "\n"))
+		_ = session.Exit(1)
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	unitName, _ := h.destination.Unit()
```

Comment:
unitName cannot be empty in case of k8s, right?

Reply:
Correct.